cli/src/lib/stubUtils.js

```diff
@@ -75,6 +79,89 @@ function stubFor(moduleName: string, fileExt?: string): string {
   return moduleStub;
 }
 
+const functionTemplate = '(%s) => any';
+const spaceByI = (i: number): string => '  '.repeat(i);
+
+let callstack = 0;
+let maxCallstack = 1;
+
+export function objectToTypedTemplate(obj: mixed, functionHeader?: string) {
+  callstack++;
+  let output = '';
+  let entries = Object.entries(obj);
+  let formatedEntries = [];
+  if (functionHeader) {
+    formatedEntries.push(spaceByI(callstack + 1) + functionHeader);
+  }
+  for (let entrie of entries) {
+    formatedEntries.push(
+      format(
+        spaceByI(callstack) + keyTypeTemplate,
+        entrie[0],
+        objectToType(entrie[1], callstack > maxCallstack ? false : true),
+      ),
+    );
+  }
+  if (formatedEntries.length > 0) {
+    output = '{\n' + formatedEntries.join('') + spaceByI(callstack) + '}';
+  } else {
+    output = '{}';
+  }
+  callstack--;
+  return output;
+}
+
+export function functionToType(fun: Function): string {
+  let output = '';
+  let raw = fun.toString();
+  let args = raw.slice(raw.indexOf('(') + 1, raw.indexOf(')'));
+  args = args.replace(/ /g, '');
+  if (args.length > 0) {
+    args = args
+      .split(',')
+      .map((el: string) => el + ': any')
+      .join(', ');
+  }
+  output = format(functionTemplate, args);
+  let functionEntries = Object.entries(fun);
+  if (functionEntries.length > 0) {
+    output = objectToTypedTemplate(
+      fun,
+      output.length > 0 ? output + ',\n' : undefined,
+    );
+  }
+  return output;
+}
+
+const keyTypeTemplate = `  %s: %s,\n`;
+export function objectToType(obj: mixed, deep: boolean = true): string {
+  let output = '';
```

Comment:
Entire function can be refactored as shown:

```
if (typeof obj === 'object') {
  return deep ? objectToTypedTemplate(obj) : 'any';
}
if (typeof obj === 'function' && deep) return functionToType(obj);
return typeof obj;
```

Haven't tested this chunk locally, so you may need to play around with the ternary return.

Reply:
I agree that current form looks bad but i would refactor it to:
```js
if(deep){
  if(typeof obj === 'object'){
    return objectToTypedTemplate(obj);
  }else if(typeof obj === 'function'){
    return functionToType(obj)
  }
}else{
  if(typeof obj === 'object'){
    return 'any';
  }
}
return typeof obj;
```
It's more open for changes and easier to understand

Reply:
Good point, checking `deep` first does make i easier to read. How about:

```
if(deep) {
  if(typeof obj === 'object') return objectToTypedTemplate(obj);
  if(typeof obj === 'function') return functionToType(obj);
}

if(typeof obj === 'object') return 'any';
return typeof obj;
```

No need for `else` statements when you return early (which is just good practice!)

Reply:
It's much better! I'll test tommorow my refactor and push it to github